core/vm/contracts_suave.go

```diff
@@ -155,6 +163,10 @@ func (c *confStoreRetrieve) RunOffchain(backend *SuaveExecutionBackend, input []
 	bidId := unpacked[0].([16]byte)
 	key := unpacked[1].(string)
 
+	return c.runImpl(backend, bidId, key)
+}
+
+func (c *confStoreRetrieve) runImpl(backend *SuaveExecutionBackend, bidId [16]byte, key string) ([]byte, error) {
```

Comment:
I wonder if this separation is useful in the general case
Would it make sense to also make the Impl functions return regular types?

Reply:
This separation of `Run` and `runImpl` was only made so that the changes to the repo were as low as possible. My idea is that the implementation of the precompiles itself goes directly to the `backendImpl` eventually where it becomes easier to test.

Reply:
Also, the `Run` and `runImpl` separation of concerns represent the two layers of the interface between the EVM and the precompiles. `Run` deals with the input/output encoding and `runImpl` with the logic. `Run` is to be deprecated in a followup PR (note it is not in use after the generated code).

The types used in `runImpl` are decided at the SuaveLib library level.

Reply:
Leaving decision up to @dmarzzz and @metachris then